src/main/java/dev/unnm3d/rediseconomy/currency/CurrencyMigration.java

```diff
@@ -0,0 +1,34 @@
+package dev.unnm3d.rediseconomy.currency;
+
+import dev.unnm3d.rediseconomy.RedisEconomyPlugin;
+import org.bukkit.Bukkit;
+
+public abstract class CurrencyMigration {
+
+    protected final RedisEconomyPlugin plugin;
+    protected final Currency currency;
+
+    public CurrencyMigration(RedisEconomyPlugin plugin, Currency currency) {
+        this.plugin = plugin;
+        this.currency = currency;
+    }
+
+    public abstract String getProvider();
+
+    protected abstract boolean setup();
+
+    protected abstract void start();
+
+    public void migrate() {
+        if (!setup()) {
+            return;
+        }
+
+        Bukkit.getConsoleSender().sendMessage("[RedisEconomy] §aMigrating from " + getProvider() + "...");
```

Comment:
Put messages in messages.yml locales!!!
And do not use sendMessage to a static call with Bukkit🤦‍♂️

Reply:
I used the same code
https://github.com/Emibergo02/RedisEconomy/blob/3338e61c6cf473a1e63d5525b0494580bf487c14/src/main/java/dev/unnm3d/rediseconomy/currency/CurrenciesManager.java#L106